source/extensions/filters/http/decompressor/decompressor_filter.h

```diff
@@ -122,12 +131,36 @@ class DecompressorFilter : public Http::PassThroughFilter,
   // Http::StreamDecoderFilter
   Http::FilterHeadersStatus decodeHeaders(Http::RequestHeaderMap&, bool) override;
   Http::FilterDataStatus decodeData(Buffer::Instance&, bool) override;
+  Http::FilterTrailersStatus decodeTrailers(Http::RequestTrailerMap&) override;
 
   // Http::StreamEncoderFilter
   Http::FilterHeadersStatus encodeHeaders(Http::ResponseHeaderMap&, bool) override;
   Http::FilterDataStatus encodeData(Buffer::Instance&, bool) override;
+  Http::FilterTrailersStatus encodeTrailers(Http::ResponseTrailerMap&) override;
 
 private:
+  struct ByteTracker {
+    ByteTracker(const Http::LowerCaseString& compressed_bytes_trailer,
+                const Http::LowerCaseString& uncompressed_bytes_trailer)
+        : compressed_bytes_trailer_(compressed_bytes_trailer),
+          uncompressed_bytes_trailer_(uncompressed_bytes_trailer) {}
+    void chargeBytes(uint64_t compressed_bytes, uint64_t uncompressed_bytes) {
```

Comment:
```suggestion
    void addByteCounts(uint64_t compressed_bytes, uint64_t uncompressed_bytes) {
```

Reply:
chargeX is a common idiom across the codebase